testApp/android/app/src/androidTest/java/com/example/reactnativedidomi/UIGettersTest.kt

```diff
@@ -36,51 +35,95 @@ class UIGettersTest {
-
-        if (needToScroll)
-            onView(withText(method.toUpperCase())).perform(ViewActions.swipeUp(), ViewActions.swipeUp(), ViewActions.swipeUp(), ViewActions.swipeUp(), ViewActions.swipeUp(), ViewActions.swipeUp(), ViewActions.swipeUp())
-        waitForDisplayed(withText("$method-OK"))
-    }
-
     @Test
     fun test_GetDisabledPurposes() {
-        testMethodCall("getDisabledPurposes", true)
+        disagreeToAll()
+
+        tapButton("getDisabledPurposes".toUpperCase())
+        assertText(ALL_PURPOSE_IDS)
     }
 
     @Test
     fun test_GetDisabledPurposeIds() {
-        testMethodCall("getDisabledPurposeIds", true)
+        disagreeToAll()
+
+        tapButton("getDisabledPurposeIds".toUpperCase())
+        assertText(ALL_PURPOSE_IDS)
     }
 
     @Test
     fun test_GetDisabledVendors() {
-        testMethodCall("getDisabledVendors", true)
+        disagreeToAll()
+
+        tapButton("getDisabledVendors".toUpperCase())
+        assertText(ALL_VENDOR_IDS)
     }
 
     @Test
     fun test_GetDisabledVendorIds() {
-        testMethodCall("getDisabledVendorIds", true)
+        disagreeToAll()
+
+        tapButton("getDisabledVendorIds".toUpperCase())
+        assertText(ALL_VENDOR_IDS)
     }
 
     @Test
     fun test_GetEnabledPurposes() {
-        testMethodCall("getEnabledPurposes", true)
+        agreeToAll()
+
+        tapButton("getEnabledPurposes".toUpperCase())
+        assertText(ALL_PURPOSE_IDS)
     }
 
     @Test
     fun test_GetEnabledPurposeIds() {
-        testMethodCall("getEnabledPurposeIds", true)
+        agreeToAll()
+
+        tapButton("getEnabledPurposeIds".toUpperCase())
+        assertText(ALL_PURPOSE_IDS)
     }
 
     @Test
     fun test_GetEnabledVendors() {
-        testMethodCall("getEnabledVendors", true)
+        agreeToAll()
+
+        tapButton("getEnabledVendors".toUpperCase())
+        assertText(ALL_VENDOR_IDS)
     }
 
     @Test
     fun test_GetEnabledVendorIds() {
-        testMethodCall("getEnabledVendorIds", true)
+        agreeToAll()
+
+        tapButton("getEnabledVendorIds".toUpperCase())
+        assertText(ALL_VENDOR_IDS)
+    }
+
+    private fun agreeToAll() {
+        tapButton("setUserAgreeToAll".toUpperCase())
+        Thread.sleep(2000L)
+        assertText("setUserAgreeToAll-OK")
+    }
+
+    private fun disagreeToAll() {
+        tapButton("setUserDisagreeToAll".toUpperCase())
+        Thread.sleep(2000L)
+        assertText("setUserDisagreeToAll-OK")
+    }
+
+    private fun tapButton(name: String) {
+        val matcher = withText(name)
+        onView(matcher).perform(ScrollToAction())
+        onView(matcher).perform(click())
+    }
+
+    private fun assertText(text: String) {
+        val matcher = withText(text)
+        onView(matcher).perform(ScrollToAction())
+    }
+
+    companion object {
+        const val ALL_VENDOR_IDS = "28,google"
+        const val ALL_PURPOSE_IDS = "cookies,create_ads_profile,geolocation_data,select_personalized_ads"
     }
 }
```

Comment:
missing blank line at the end of the file

Reply:
done